### PR TITLE
INN-1521: Add Apps Page

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description

- Add Apps page
Screenshot 2023-07-10 at 15 30 48

To do:
- [x] Wire get apps data
- [x] Ability to manually add App
- [x] Ability to delete App
- [x] Empty & load states
- [x] Enable page in the navBar
- [ ] Optimist render
- [x] Navbar notice

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [X] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/src/components/App/AppCardStep.tsx [hunk 1]
```diff
@@ -38,10 +38,10 @@ export default function AppCardStep({
...
         <Transition
-          enter="transition-opacity duration-300"
+          enter="transition-opacity duration-200"
           enterFrom="opacity-0"
           enterTo="opacity-100"
-          leave="transition-opacity duration-300"
+          leave="transition-opacity duration-200"
           leaveFrom="opacity-100"
           leaveTo="opacity-0"
         >
```
Comment (IGassmann) [inline ui/src/components/App/AppCardStep.tsx:47]: **suggestion (ux)**: the animation feels a bit clunky to me:

https://github.com/inngest/inngest/assets/4291707/d4edbe83-4bc8-4c70-b2b7-6b432d660847

An animation in this style might fit better:

https://github.com/inngest/inngest/assets/4291707/b26b6d50-4bde-4edd-a58c-8d85fdc5fcac

**Review by IGassmann — CHANGES_REQUESTED** [review]
Thanks for implementing this. I'm looking forward to all those changes! :)

I've only had a chance to review the UX for now, as there are already a lot of comments. I'll take a look at the code later.

For future PRs, I think it would be a good idea to make them smaller. This first pass alone took me an hour and a half to review.

**Comment by anafilipadealmeida** [comment]
> Thanks for implementing this. I'm looking forward to all those changes! :)
> 
> I've only had a chance to review the UX for now, as there are already a lot of comments. I'll take a look at the code later.
> 
> For future PRs, I think it would be a good idea to make them smaller. This first pass alone took me an hour and a half to review.

Thanks for reviewing this @IGassmann! A couple of notes:
- I was expecting the UX review to happen outside the scope of this PR, and along the way. That's why we've been discussing UX and UI in the Discord channel for the project instead. I'll move some of those there, as Ed is the best one to answer some of those concerns. What I was expecting here was the code review.
- Yes, ideally the PRs would be smaller, but given how the backend code was being built all at the same time, this was way faster than waiting for things in multiple PRs simultaneously.

File: ui/src/components/App/AppCard.tsx [hunk 2]
```diff
@@ -1,124 +1,213 @@
...
-              </div>
+              </form>
               <a
                 className="text-indigo-400 flex items-center gap-2 cursor-pointer w-fit"
                 onClick={() => dispatch(showDocs('/sdk/serve'))}
               >
                 Connecting to the Dev Server
-                <IconArrowTopRightOnSquare />
+                <IconBook />
               </a>
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/App/AppCard.tsx:188]: That's a new feature, out of the scope of this PR. We have a different [issue](https://linear.app/inngest/issue/INN-1523/replace-current-redux-store-using-next-js-for-routing-and-local-state) in the linear project to add routes now that we have Nextjs, as I agree that this makes the experience annoying.

File: ui/src/components/App/AppCard.tsx [hunk 3]
```diff
@@ -136,7 +225,7 @@ export default function AppCard({ app }: AppCardProps) {
           }
           expandedContent={
             <>
-              {(app.status !== 'connected' || app.functionCount < 1) && (
+              {(!app.connected || app.functionCount < 1) && (
                 <>
                   <p className="pb-4 text-slate-400">
                     There are currently no functions registered at this url.
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/App/AppCard.tsx:231]: I assumed that with the icon there it was pretty obvious that the element is clickable and wouldn't require a cursor change, but I'll add it to make it clearer 🙂

File: ui/src/components/Navbar/NavbarLink.tsx [hunk 4]
```diff
@@ -34,6 +37,7 @@ export default function NavebarLink({
     >
       {icon}
       {tabName}
+      {typeof badge === 'number' && <Badge kind={hasError ? 'error' : 'outlined'}>{badge.toString()}</Badge>}
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/Navbar/NavbarLink.tsx:40]: Moving this to discord!

File: ui/src/components/Badge.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  children: React.ReactNode;
+  className?: string;
+  kind?: 'outlined' | 'error';
+}) {
+  const classNames = `text-xs leading-3 border rounded-md  box-border py-1.5 px-2 flex items-center gap-1 ${kindStyles[kind]} ${className}`;
+
+  return (
+    <span className={classNames}>
+      {kind === 'error' && (
+        <IconExclamationTriangleSolid className="text-rose-400 w-3 h-3" />
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/Badge.tsx:22]: Moving this to discord!

File: ui/src/components/App/AddAppModal.tsx [hunk 6]
```diff
@@ -1,30 +1,108 @@
...
+              className={classNames(
+                'min-w-[420px] bg-slate-800 rounded-md text-slate-300 py-2 px-4 outline-2 outline-indigo-500 focus:outline',
+                isUrlInvalid && inputUrl.length > 0 && 'pr-8 outline-rose-400'
+              )}
+              placeholder="https://example.com/api/inngest"
+              value={inputUrl}
+              onChange={handleChange}
+            />
+            {isUrlInvalid && inputUrl.length > 0 && (
+              <IconExclamationTriangleSolid className="absolute top-2/4 right-2 -translate-y-2/4 text-rose-400" />
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/App/AddAppModal.tsx:95]: Moving this to discord!

File: ui/src/components/App/AppCard.tsx [hunk 7]
```diff
@@ -1,124 +1,213 @@
...
+                {app.connected && app.functionCount > 0 ? (
                   <>
                     {<IconAppStatusCompleted />}
                     {app.functionCount} Functions registered
                   </>
-                ) : app.status !== 'connected' ? (
+                ) : !app.connected ? (
                   <>{<IconAppStatusDefault />}No Functions Found</>
                 ) : (
                   <>{<IconAppStatusFailed />}No Functions Found</>
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/App/AppCard.tsx:206]: Moving this to discord!

File: ui/src/components/App/AddAppModal.tsx [hunk 8]
```diff
@@ -1,30 +1,108 @@
...
+      });
+      toast.success('The app was successfully added.');
+      console.log('Created app:', response);
+    } catch (error) {
+      toast.error('The app could not be created: ${error}.');
+      console.error('Error creating app:', error);
+    }
+    onClose();
+    // To do: add optimistic render in the list
+  }
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/components/App/AddAppModal.tsx:50]: Backend issue, addressed in: https://github.com/inngest/inngest/pull/500

File: ui/src/views/AppList.tsx
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply ui/src/views/AppList.tsx:1]: Backend issue, addressed in: https://github.com/inngest/inngest/pull/500

**Review by tonyhb — APPROVED** [review]

**Pull request merged**